web3swift/src/Client/Models/EthereumAddress.swift

```diff
@@ -4,37 +4,61 @@
 //
 
 import Foundation
+import BigInt
 
 public struct EthereumAddress: Codable, Hashable {
-    public let value: String
+    @available(*, deprecated, message: "Shouldn't rely on the actual String representation. Use asString() instead to get an unformatted representation")
+    public var value: String {
+        raw
+    }
+    private let raw: String
     public static let zero: Self = "0x0000000000000000000000000000000000000000"
 
     public init(_ value: String) {
-        self.value = value.lowercased()
+        self.raw = value.lowercased()
```

Comment:
Do we need `.lowercased()` ?  Same question for the decode implementation. If we used `RawRepresentable<String>` the `rawValue` would always maintain its initial case.

Reply:
keeping the assumptions as before. I  suspect that `lowercased()` was introduced when there were issues with comparision, which shouldn't be neede anymore. But I want to keep the same behaviour as before for now.